### PR TITLE
block adapter remove dependency in s3 for multipart struct

### PR DESCRIPTION
**Pull request by nopcoder**
Waiting for #2684 - do not merge

### CODE REVIEW

File: pkg/block/adapter.go [hunk 1]
```diff
@@ -4,15 +4,21 @@ import (
...
 	"io"
 	"net/http"
-
-	"github.com/aws/aws-sdk-go/service/s3"
 )
 
+// MultipartPart single multipart information
+type MultipartPart struct {
+	ETag       string
+	PartNumber int64
```
Comment (arielshaqed) [inline pkg/block/adapter.go:12]: Any point in `int64` here?  The [docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html#mpu-process) say it's a number between 1 and 10_000...

File: pkg/block/s3/adapter.go [hunk 2]
```diff
@@ -563,20 +563,30 @@ func (a *Adapter) AbortMultiPartUpload(ctx context.Context, obj block.ObjectPoin
 	return err
 }
 
+func convertFromBlockMultipartUploadCompletion(multipartList *block.MultipartUploadCompletion) *s3.CompletedMultipartUpload {
+	var parts []*s3.CompletedPart
```
Comment (arielshaqed) [inline pkg/block/s3/adapter.go:567]:
```suggestion
	parts := make([]*s3.CompletedPart, 0, len(multipartList.Part))
```

**Review by arielshaqed — APPROVED** [review]
Great cleanup, thanks!

File: pkg/block/adapter.go (hunk 1 above)
Comment (nopcoder) [reply pkg/block/adapter.go:12]: no point - kept the AWS types - but it will be easier to use int, we do not use code-generators yet.

**Comment by nopcoder** [comment]
Transformed adapter to work with `int` instead of `int64` for multipart

**Review by arielshaqed — COMMENTED** [review]
Cool, stillLGM...

**Pull request merged**